src/index.ts

```diff
@@ -1,87 +1,54 @@
 /*
- * Bundle: Meili
- * Project: Meili - Javascript API
+ * Bundle: Meilisearch
+ * Project: Meilisearch - Javascript API
  * Author: Quentin de Quelen <quentin@meilisearch.com>
- * Copyright: 2019, Meili
+ * Copyright: 2019, Meilisearch
  */
 
 'use strict'
 
-import instance, { AxiosInstance } from 'axios'
-
+import MeiliAxiosWrapper from './meili-axios-wrapper'
 import { Indexes } from './indexes'
 import * as Types from './types'
 
-interface Config {
-  host: string
-  apiKey?: string
-}
-
-class Meili {
-  baseURL: string
-  apiKey?: string
-  instance: AxiosInstance
-
-  constructor(config: Config) {
-    this.baseURL = config.host
-    this.apiKey = config.apiKey
-
-    if (config.apiKey) {
-      this.instance = instance.create({
-        baseURL: this.baseURL,
-        headers: {
-          'X-Meili-API-Key': config.apiKey,
-        },
-      })
-    } else {
-      this.instance = instance.create({
-        baseURL: this.baseURL,
-      })
-    }
-    this.instance.interceptors.response.use((response) => response.data)
-    this.instance.interceptors.request.use((request) => {
-      if (request.data !== undefined) {
-        return {
-          ...request,
-          data: JSON.stringify(request.data),
-        }
-      }
-
-      return request
-    })
+class Meilisearch extends MeiliAxiosWrapper {
+  config: Types.Config
+  constructor(config: Types.Config) {
+    super(config)
+    this.config = config
   }
 
   /**
    * Return an Index instance
-   * @memberof Meili
+   * @memberof Meilisearch
    * @method Index
    */
   getIndex(indexUid: string): Indexes {
-    return new Indexes(this.instance, indexUid)
+    return new Indexes(this.config, indexUid)
   }
 
   /**
    * List all indexes in the database
-   * @memberof Meili
+   * @memberof Meilisearch
    * @method listIndexes
    */
   listIndexes(): Promise<object[]> {
     const url = '/indexes'
 
-    return this.instance.get(url)
+    return this.get(url)
   }
 
   /**
    * Create a new index
-   * @memberof Meili
+   * @memberof Meilisearch
    * @method createIndex
    */
   createIndex(
     data: Types.CreateIndexRequest
   ): Promise<Types.CreateIndexResponse> {
     const url = `/indexes`
 
-    return this.instance.post(url, data)
+    return this.post(url, data).then((res: any) => res)
```

Comment:
why this one must have a `.then` with an `any`. This is this line who forces you to add `allow-any: false` in the linter?

Reply:
This is a mistake. I will remove it. No it's a lot of lines in the axios wrapper that uses `any`.